smack-integration-test/src/main/java/org/jivesoftware/smackx/mam/MamIntegrationTest.java

```diff
@@ -0,0 +1,62 @@
+/**
+ *
+ * Copyright 2016 Fernando Ramirez
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jivesoftware.smackx.mam;
+
+import org.igniterealtime.smack.inttest.AbstractSmackIntegrationTest;
+import org.igniterealtime.smack.inttest.SmackIntegrationTest;
+import org.igniterealtime.smack.inttest.SmackIntegrationTestEnvironment;
+import org.jivesoftware.smack.packet.Message;
+import org.jivesoftware.smackx.mam.MamManager.MamQueryResult;
+import org.junit.Assert;
+import org.jxmpp.jid.Jid;
+import org.jxmpp.jid.impl.JidCreate;
+
+public class MamIntegrationTest extends AbstractSmackIntegrationTest {
+
+    private final MamManager mamManager;
+
+    public MamIntegrationTest(SmackIntegrationTestEnvironment environment) {
+        super(environment);
+        mamManager = MamManager.getInstanceFor(conOne);
+    }
+
+    @SmackIntegrationTest
+    public void mamTest() throws Exception {
+
+        if (!mamManager.isSupportedByServer()) {
+            return;
+        }
+
+        Jid userOne = JidCreate.from(conOne.getUser().toString().split("/")[0]);
+        Jid userTwo = JidCreate.from(conTwo.getUser().toString().split("/")[0]);
+
+        MamQueryResult mamQueryResult = mamManager.queryArchive(userTwo);
+
+        if (mamQueryResult.forwardedMessages.isEmpty()) {
```

Comment:
A minimal integration test for MAM looks like:
1. useOne sends *one* message two userTwo
2. userTwo queries its archive
3. ensure that the last message in the archive it is exactly the message send (compare at least the stanza id and the body content)

Of course it would be really great to have at least one more sophisticated integration test, e.g. something that checks the functionality with more than one message.